packages/database/src/migrations/20210803035205-CovidSamoaAddVillageEntityRelations-modifies-data.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+import { villageRelations } from './migrationData/20210803014704-CovidSamoaAddSubDistrictEntities/Entities - Village Update 20210802';
+import { arrayToDbString, findSingleRecord, generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  // console.log('villageRelations: ', villageRelations);
+  const villageCodes = Object.keys(villageRelations);
+  const subDistrictCodes = Object.values(villageRelations);
+
+  const samoaVillageRows = (
+    await db.runSql(`
+      select code, id from entity where "code" in (${arrayToDbString(villageCodes)});
+    `)
+  ).rows;
+  const samoaVillageIds = Object.assign({}, ...samoaVillageRows.map(x => ({ [x.code]: x.id })));
```

Comment:
Oh this is a nifty trick!!! I usually use a reduce in this scenario but yours seems quicker!
Gotta be careful tho with spread operator in function arguments, can lead to maximum call stack limit exceeded when dealing with large arrays (10K +)

Reply:
Ah good to know!